core/tee/tee_fs_common.c

```diff
@@ -37,19 +40,939 @@
 
 static struct handle_db fs_handle_db = HANDLE_DB_INITIALIZER;
 
+struct block_operation_args {
+	struct tee_fs_fd *fdp;
+	tee_fs_off_t offset;
+	int block_num;
```

Comment:
As I see it this only uses positive numbers and therefore I suggest to change it to `size_t` instead.